lib/practice_exercises.rb

```diff
@@ -1,13 +1,35 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def is_palindrome(string)
```

Comment:
👍 
Technically because you make another string with `string.gsub(...).downcase` you have O(n) space complexity.